extensions/replication/queueProcessor/task.js

```diff
@@ -169,23 +169,31 @@ function initAndStart(zkClient) {
         log.info('management init done');
 
         const bootstrapList = config.getBootstrapList();
+        const siteNames = bootstrapList.map(i => i.site);
 
-        const destConfig = Object.assign({}, repConfig.destination);
-        destConfig.bootstrapList = bootstrapList;
+        // initialize per site destination configs
+        const destConfig = {};
+        siteNames.forEach(site => {
+            destConfig[site] = config.getReplicationSiteDestConfig(site);
+            destConfig[site].bootstrapList = bootstrapList;
+        });
 
         config.on('bootstrap-list-update', () => {
-            destConfig.bootstrapList = config.getBootstrapList();
-
+            const updatedBootstrapList = config.getBootstrapList();
             const activeSites = Object.keys(activeQProcessors);
-            const updatedSites = destConfig.bootstrapList.map(i => i.site);
+            const updatedSites = updatedBootstrapList.map(i => i.site);
             const allSites = [...new Set(activeSites.concat(updatedSites))];
 
             async.each(allSites, (site, next) => {
                 if (updatedSites.includes(site)) {
+                    if (!destConfig[site]) {
+                        destConfig[site] = config.getReplicationSiteDestConfig(site);
+                    }
+                    destConfig[site].bootstrapList = updatedBootstrapList;
```

Comment:
boostrapList already set in `getReplicationSiteDestConfig()`, so either it is an else:

```suggestion
                if (!destConfig[site]) {
                    destConfig[site] = config.getReplicationSiteDestConfig(site);
                }
                else {
                    destConfig[site].bootstrapList = updatedBootstrapList;
                }
```

or maybe we should always update the site's configuration:

```suggestion
                destConfig[site] = config.getReplicationSiteDestConfig(site);
```

Reply:
We shouldn't update the whole site's configuration every time as that will make the pre-existent QueueProcessors loose reference to it as it's a new object. I added an else case.